### PR TITLE
[GOBBLIN-296] Kafka json source and writer

### PR DESCRIPTION
**Pull request by zxcware**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title.
    - https://issues.apache.org/jira/browse/GOBBLIN-296

### Description
- [x] Here are some details about my PR:
  - Add a json source and writer for kafka 09: `Kafka09JsonSource` and `Kafka09JsonObjectWriterBuilder`
  - Move common gson ser/de logic to gobblin-kafka-common module

### Tests
- [x] My PR adds the following unit tests
  - `Kafka09JsonIntegrationTest`

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by zxcware** [comment]
@htran1 Please review. Thanks.

File: gobblin-modules/gobblin-kafka-09/src/main/java/org/apache/gobblin/source/extractor/extract/kafka/Kafka09JsonSource.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.source.extractor.extract.kafka;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.gobblin.configuration.SourceState;
```
Comment (htran1) [inline gobblin-modules/gobblin-kafka-09/src/main/java/org/apache/gobblin/source/extractor/extract/kafka/Kafka09JsonSource.java]: Gobblin should be imported after com according to the Gobblin conventions.

File: gobblin-modules/gobblin-kafka-09/src/test/java/org/apache/gobblin/kafka/Kafka09JsonIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.kafka;
+
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.util.List;
+
+import org.apache.gobblin.configuration.ConfigurationKeys;
```
Comment (htran1) [inline gobblin-modules/gobblin-kafka-09/src/test/java/org/apache/gobblin/kafka/Kafka09JsonIntegrationTest.java]: Special import order for org.apache.gobblin is after java, org, com.

File: gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/kafka/writer/AbstractKafkaDataWriterBuilder.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.kafka.writer;
+
+import java.io.IOException;
+import java.util.Properties;
+
+import org.apache.gobblin.configuration.State;
```
Comment (htran1) [inline gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/kafka/writer/AbstractKafkaDataWriterBuilder.java]: org.apache.gobblin should be after com.

File: gobblin-runtime/src/main/java/org/apache/gobblin/runtime/util/MultiWorkUnitUnpackingIterator.java [hunk 4]
```diff
@@ -31,29 +31,59 @@
 @RequiredArgsConstructor
 public class MultiWorkUnitUnpackingIterator implements Iterator<WorkUnit> {
   private final Iterator<WorkUnit> workUnits;
+
+  /** The iterator for {@link #nextWu} if it's a {@link MultiWorkUnit} */
```
Comment (htran1) [inline gobblin-runtime/src/main/java/org/apache/gobblin/runtime/util/MultiWorkUnitUnpackingIterator.java]: Why is this file being changed?

Comment (zxcware) [reply gobblin-runtime/src/main/java/org/apache/gobblin/runtime/util/MultiWorkUnitUnpackingIterator.java]: This is to fix a bug where the iterator has tailing `MultiWorkUnit` with empty work unit list.

**Pull request closed without merging**